src/lib/mina_base/account_update.ml

```diff
@@ -955,83 +955,52 @@ module Account_precondition = struct
   [%%versioned
   module Stable = struct
     module V1 = struct
-      type t =
-            Mina_wire_types.Mina_base.Account_update.Account_precondition.V1.t =
-        | Full of Zkapp_precondition.Account.Stable.V2.t
-        | Nonce of Account.Nonce.Stable.V1.t
-        | Accept
+      type t = Zkapp_precondition.Account.Stable.V2.t
       [@@deriving sexp, yojson, hash]
 
-      let to_latest = Fn.id
-
-      let to_full = function
-        | Full s ->
-            s
-        | Nonce n ->
-            { Zkapp_precondition.Account.accept with
-              nonce = Check { lower = n; upper = n }
-            }
-        | Accept ->
-            Zkapp_precondition.Account.accept
+      let (_ :
```

Comment:
Rather than having this explicit type equality check here, why not just specify the above type to just be `Mina_wire_types.Mina_base.Account_update.Account_precondition.V1.t`? The type checker will ensure that this type unifies with `Zkapp_precondition.Account.Stable.V2.t` if needed.

Reply:
It's useful know that this type is the same as the other two types, not for the type checker, but for humans.

It might make sense to swap them between the type definition and the equality assertion.